File: firebase/functions/lib/events/calendar/calendar_feed_rss.dart
import 'dart:async';
import 'dart:io';

import 'abstract_calendar_feed.dart';
import 'rss_util.dart';
import 'package:data_models/community/community.dart';

/// Generate and return an RSS feed of upcoming events for a given space. This function expects
/// the space id to be present as the second request path parameter, as in '/space/[space_id]/rss'.
class CalendarFeedRss extends AbstractCalendarFeed {
  @override
  final String functionName = 'CalendarFeedRss';

  @override
  Future<String> generateData({required Community community}) {
    return rssUtil.getRssForUpcomingEvents(community: community);
  }

  @override
  ContentType getContentType() {
    return ContentType('application', 'xml');
  }
}

File: firebase/functions/src/events/calendar/calendar_feed_rss.ts
import { AbstractCalendarFeed } from './abstract_calendar_feed';
import { rssUtil } from './rss_util';

export class CalendarFeedRss extends AbstractCalendarFeed {
  readonly functionName = 'CalendarFeedRss';

  async generateData({ community }: { community: Record<string, unknown> }): Promise<string> {
    return rssUtil.getRssForUpcomingEvents({ community });
  }

  getContentType(): string {
    return 'application/xml';
  }
}
